api/modules/Catalog/Catalog.Infrastructure/Endpoints/v1/DeleteProductEndpoint.cs

```diff
@@ -0,0 +1,21 @@
+﻿using FSH.Framework.Infrastructure.Auth.Policy;
+using FSH.WebApi.Catalog.Application.Products.Delete.v1;
+using MediatR;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Routing;
+
+namespace FSH.WebApi.Catalog.Infrastructure.Endpoints.v1;
+public static class DeleteProductEndpoint
+{
+    internal static RouteHandlerBuilder MapProductDeleteEndpoint(this IEndpointRouteBuilder endpoints)
+    {
+        return endpoints.MapDelete("/{id:guid}", (Guid id, ISender mediator) => mediator.Send(new DeleteProductCommand(id)))
```

Comment:
can you make this an async/await operation. refer to the update product endpoint. Maybe you can do this for other product endpoints too, if they are not already changed